admin/installer/cloudformation.py

```diff
@@ -70,19 +70,29 @@ def _sibling_lines(filename):
 # Keys corresponding to CloudFormation user Inputs.
 access_key_id_param = template.add_parameter(Parameter(
     "AmazonAccessKeyID",
-    Description="Your Amazon AWS access key ID",
+    Description="Your Amazon AWS access key ID (mandatory)",
     Type="String",
+    NoEcho=True,
+    AllowedPattern="[\w]+",
+    MinLength="16",
+    MaxLength="32",
 ))
 secret_access_key_param = template.add_parameter(Parameter(
     "AmazonSecretAccessKey",
-    Description="Your Amazon AWS secret access key.",
+    Description="Your Amazon AWS secret access key (mandatory)",
     Type="String",
+    NoEcho=True,
+    MinLength="1",
 ))
 keyname_param = template.add_parameter(Parameter(
     "EC2KeyPair",
     Description="Name of an existing EC2 KeyPair to enable SSH "
-                "access to the instance",
+                "access to the instance (mandatory)",
     Type="String",
```

Comment:
:question: Perhaps we could add an *invalid* default value? It seems like it might be worth giving it another try since the validation should quickly report whether the named key is available in this region and I think @robhaswell  and someone else already stumbled across that problem.